sympy/functions/elementary/complexes.py

```diff
@@ -658,10 +658,13 @@ def _eval_power(self, exponent):
 
     def _eval_nseries(self, x, n, logx, cdir=0):
         from sympy.functions.elementary.exponential import log
+        from sympy.series.order import Order
         direction = self.args[0].leadterm(x)[0]
         if direction.has(log(x)):
             direction = direction.subs(log(x), logx)
         s = self.args[0]._eval_nseries(x, n=n, logx=logx)
+        if direction.is_extended_real is False:
+            return im(s.removeO()) + Order(x**n, x)
         return (sign(direction)*s).expand()
```

Comment:
The tests look good though I'll have to think through this. I remember while I was appraoching this a couple of years back I ended up justifying this `return (s/sign(direction)).expand()` to be the correct return statement finally.